### PR TITLE
Fleet qa/65 improve deletion and more

### PR DESCRIPTION
**Pull request by mmartin24**
Implementation of https://github.com/rancher/fleet-e2e/issues/65

# Done:
- Created delete function for leftovers of Gitrepos either in default or local
- Corrected function that left logs of unwanted info on runners
- Added extra check of the element in the creating repo process to help minimize duplication of repos after clicking `Create` button
- Refactor delete function to include locator of check deleted repo page
- Others refactors
- Added resource check function
- Major refactor on tests for better reading

# Test done
https://github.com/rancher/fleet-e2e/actions/runs/8328879488/job/22789937537

### CODE REVIEW

File: tests/cypress/e2e/unit_tests/p0_fleet.spec.ts [hunk 1]
```diff
@@ -63,7 +60,6 @@ describe('Fleet Deployment Test Cases', () => {
 
       // Delete all git repos
       cy.deleteAll();
```
Comment (sbulage) [inline tests/cypress/e2e/unit_tests/p0_fleet.spec.ts:62]: Can this be used: `cy.deleteAllFleetRepos();` here?

File: tests/cypress/e2e/unit_tests/p0_fleet.spec.ts [hunk 2]
```diff
@@ -124,7 +116,6 @@ describe('Fleet Deployment Test Cases', () => {
       cy.accesMenuSelection('Continuous Delivery', 'Git Repos');
       cy.verifyTableRow(0, repoName, ' ')
       cy.deleteAll();
```
Comment (sbulage) [inline tests/cypress/e2e/unit_tests/p0_fleet.spec.ts:118]: ditto and other places too

**Review by sbulage — CHANGES_REQUESTED** [review]
Few comments, rest looks good to me.

File: tests/cypress/e2e/unit_tests/p0_fleet.spec.ts [hunk 3]
```diff
@@ -139,9 +130,7 @@ describe('Fleet Deployment Test Cases', () => {
       const userOrPublicKey = Cypress.env("bitbucket_private_user");
       const pwdOrPrivateKey = Cypress.env("bitbucket_private_pwd");
   
-      // // Click on the Continuous Delivery's icon
-      cypressLib.accesMenu('Continuous Delivery');
-      cypressLib.accesMenu('Git Repos');
+      // Click on the Continuous Delivery's icon
```
Comment (sbulage) [inline tests/cypress/e2e/unit_tests/p0_fleet.spec.ts:133]: This comment is not relevant here, as you're changing namespace...

File: tests/cypress/e2e/unit_tests/p0_fleet.spec.ts (hunk 1 above)
Comment (mmartin24) [reply tests/cypress/e2e/unit_tests/p0_fleet.spec.ts:62]: here yes, but I always thought that if having to navigate to one more place better to use the one in same screen.
Other places needs re-work. also with deletion before all.
I will work on it though

**Comment by thehejik** [comment]
Please don't forget switch to `fleet-default` namespace.

**Review by sbulage — APPROVED** [review]
LGTM :+1:

**Pull request merged**